### PR TITLE
MSATP - Updated table names and column names to support new API schema 

### PR DESCRIPTION
**Pull request by barvhaim**
Schema updated on Dec, 2019 - https://techcommunity.microsoft.com/t5/microsoft-defender-for-endpoint/advanced-hunting-data-schema-changes/ba-p/1043914
In order to support this changes we've updated the connector

### CODE REVIEW

File: stix_shifter_modules/msatp/stix_translation/json/to_stix_map.json [hunk 1]
```diff
@@ -735,22 +783,37 @@
...
     ],
-    "ComputerName": {
-      "key": "x_msatp.computer_name",
-      "cybox": false
-    },
-    "MachineId": {
-      "key": "x_msatp.machine_id",
-      "cybox": false
+    "DeviceId": {
+      "key": "x-oca-asset.id",
```
Comment (mdazam1942) [inline stix_shifter_modules/msatp/stix_translation/json/to_stix_map.json:787]: change this to `x-oca-asset.deviceid`. stix-validator throws invalid object errors

`id` is standard stix property so it has to match certain regex pattern.

File: stix_shifter_modules/msatp/configuration/config.json [hunk 2]
```diff
@@ -37,6 +37,10 @@
             },
             "clientSecret": {
                 "type": "password"
+            },
+            "access_token": {
```
Comment (mdazam1942) [inline stix_shifter_modules/msatp/configuration/config.json:41]: why did you add `access_token` in the configuration?

File: stix_shifter_modules/msatp/stix_translation/json/to_stix_map.json [hunk 3]
```diff
@@ -212,25 +212,25 @@
...
-        "references": ["src_mac"]
+        "references": "mac"
       },
       {
-        "key": "ipv6-addr.resolves_to_refs",
-        "object": "src_ip",
-        "references": ["src_mac"]
+        "key": "network-traffic.src_ref",
+        "object": "nt",
+        "references": "src_ip"
```
[the remark this answers is not in the document]
Comment (barvhaim) [reply stix_shifter_modules/msatp/stix_translation/json/to_stix_map.json:226]: @mdazam1942 not sure what do you mean here it's just single `src_ref`

File: stix_shifter_modules/msatp/stix_translation/json/to_stix_map.json [hunk 4]
```diff
@@ -212,25 +212,25 @@
...
         "key": "mac-addr.value",
-        "object": "src_mac",
+        "object": "mac",
         "transformer": "FormatMac"
       },
       {
         "key": "ipv4-addr.resolves_to_refs",
         "object": "src_ip",
-        "references": ["src_mac"]
+        "references": "mac"
```
[the remark this answers is not in the document]
Comment (barvhaim) [reply stix_shifter_modules/msatp/stix_translation/json/to_stix_map.json:221]: fixed

File: stix_shifter_modules/msatp/stix_translation/json/to_stix_map.json (hunk 1 above)
Comment (barvhaim) [reply stix_shifter_modules/msatp/stix_translation/json/to_stix_map.json:787]: fixed

File: stix_shifter_modules/msatp/configuration/config.json (hunk 2 above)
Comment (barvhaim) [reply stix_shifter_modules/msatp/configuration/config.json:41]: The way the original connector works is when you set the tenant, clientId and clientSecret, it will generate an access token and place it in the configuration under `access_token` field automatically. If you don't have this field in the config, when the config is being validated after changed by the connector it will raise an error that the structure is incorrect, thus I added this field as optional in order to bypass this structure change validation error.

Comment (mdazam1942) [reply stix_shifter_modules/msatp/configuration/config.json:41]: Yeah we are aware of this issue. this is a bug in cp4s 1.6 and earlier. It is fixed in 1.7. so you don't need to add `access_token` as optional anymore

File: stix_shifter_modules/msatp/stix_translation/json/to_stix_map.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (mdazam1942) [reply stix_shifter_modules/msatp/stix_translation/json/to_stix_map.json:226]: sorry my bad. didn't notice you changed it to src_ref

File: stix_shifter_modules/msatp/configuration/config.json (hunk 2 above)
Comment (barvhaim) [reply stix_shifter_modules/msatp/configuration/config.json:41]: @mdazam1942 ok I removed it

**Review by mdazam1942 — APPROVED** [review]
